### PR TITLE
Fix JSON cleaning step

### PR DESCRIPTION
**Pull request by omarkmu**
Used an iteration-based approach to renaming keys and trimming values. Definitely less concise, but should help to avoid any future issues with cleaning the JSON.

@Fox-Islam

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -66,24 +76,70 @@ function sendRequest(parameters: any) {
...
+            elem = container[key];
+
+            // handle the tab and newline characters
+            if (Array.isArray(elem) && elem.length === 1 && typeof elem[0] === 'string') {
+                // this branch can be removed when explicitArray option is used
+                elem[0] = elem[0].trim();
+            } else if (typeof elem === 'string') {
+                elem = elem.trim();
+                container[key] = elem;
+            }
```
Comment (Fox-Islam) [inline src/index.ts]: For some of these `if` statements, do you think it's okay to move them out into separate functions? 
My preference is to have fewer comments by having more functions with descriptive names if it's possible, so in this case doing something like:
```javascript
function handleTabAndNewline(...) {
    if {
        ...
        return
    }
    ...
}
```
The other preference I have is fewer `else` blocks by doing early `return`s in functions like that but that's less important to me than having descriptive function names, I think 😅

**Review by Fox-Islam — COMMENTED** [review]
Sorry this took so long, it's been really busy here! Thank you for working through this, the code looks good! Only had one comment, let me know what you think!

File: src/index.ts (hunk 1 above)
Comment (omarkmu) [reply src/index.ts]: @Fox-Islam I've added the changes you mentioned to the if statements involving processing the key and value, but I've left the part which adds elements to the stack alone. Would you prefer that if statement to be extracted as well?

Comment (Fox-Islam) [reply src/index.ts]: @omarkmu Ah, wonderful! No, thank you, I think it looks good, happy to merge this!

**Pull request merged**